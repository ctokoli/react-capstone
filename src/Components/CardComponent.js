import PropTypes from 'prop-types';

const CardComponent = ({ item }) => (
  <div className="card-item">
    <div className="active-data">
      <div className="top">
        <h3>
          {item.country}
        </h3>
      </div>
    </div>
    <div className="inactive-data">
      <h5>
        Total Population
      </h5>
      <h6>{item.population}</h6>
    </div>
  </div>
);

CardComponent.propTypes = {
  item: PropTypes.array.isRequired,
};
export default CardComponent;
