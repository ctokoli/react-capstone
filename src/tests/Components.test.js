import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import CardComponent from '../Components/CardComponent';
import DetailsCard from '../Components/DetailsCardComponent';

describe('Test for the components', () => {
  it('should render the cardComponent', () => {
    const item = [];
    const card = renderer.create(
      <CardComponent
        item={item}
      />,
    ).toJSON();
    expect(card).toMatchSnapshot();
  });
  it('should the cardComponent', () => {
    const item = [];
    const key = 'value';
    render(<DetailsCard key={key} item={item} />);
    const headings = screen.getByText('Total Population');
    expect(headings).toBeInTheDocument();
  });
});
