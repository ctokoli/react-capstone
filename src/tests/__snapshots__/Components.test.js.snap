// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the components should render the cardComponent 1`] = `
Array [
  <div>
    <div
      className="top"
    >
      <h3 />
    </div>
  </div>,
  <div>
    <h6>
      Total Recovered
    </h6>
    <h6 />
  </div>,
]
`;
